modules/chapter1/pages/section2.adoc

```diff
@@ -1,3 +1,532 @@
+* *Step* - (AKA _pipeline component_) is a self-contained bundle of code that represents an execution stage in the pipeline.
+* *Experiment* - is a logical grouping of runs where you can try different pipeline configurations.
+* *Run* - is a _single_ execution of a pipeline whereas a recurring run is a scheduled, repeatable execution of a pipeline.
+
+[NOTE]
+====
+Pipleines are a basically an execution graph of tasks, this graph is commonly known as a _DAG_ (Directed Acyclic Graph).
+A DAG is a directed graph with no cycles aka no direct loops.
+====
+
+OpenShift AI offers two out of the box mechanisms to build Data Science Pipelines.
+
+The first mechanism is the Elyra Jupyter notebook addon which provides a visual editor for creating pipelines via orchestration of Jupyter notebooks. 
+
+The second mechanism and the one discussed here is using Data Science Pipelines (DSP) based on the upstream KubeflowPipelines V1. With DSP, pipelines are built using python scripts using the KFPV1 python SDK. Once built, the python pipeline is submitted to the DSP runtime to be scheduled for execution.
+
+[NOTE]
+====
+There is work in progress to migrate DSP to Kubeflow Pipeline V2. At the time of writing this is still a work in progress and we won't cover it here.
+====
+
+In OpenShift AI we use the *_Tekton_* runtime to execute pipelines. With this there is an additional step to be performed the python script needs to be compiled into a Tekton definition before being submitting to the runtime
+
+In OpenShift AI the Data Science Pipeline runtime consists of the following components:
+
+* A Data Science Pipeline Server container. 
+* A MariaDB for storing pipeline definitions and results.
+* A Pipeline scheduler for scheduling pipeline runs.
+* A Persistent Agent to record the set of containers that executed as well as their inputs and outputs.
+
+Steps in the pipeline are executed as ephemeral pods (one per step).
+
+[NOTE]
+====
+DS Pipeline in OpenShift AI are managed by the `data-science-pipelines-operator-controller-manager` operator in the `redhat-ods-applications` namespace. The CRD is an instance of _datasciencepipelinesapplications.datasciencepipelinesapplications.opendatahub.io_
+====
+
+.DataScience Pipeline pods
+[source,cmd]
+----
+ ~/oc get pods -n redhat-ods-applications
+NAME                                                              READY   STATUS    RESTARTS   AGE
+ds-pipeline-persistenceagent-pipelines-definition-6d6676d58vzss   1/1     Running   0          9m49s
+ds-pipeline-pipelines-definition-7c7dd56b4d-qfz8t                 2/2     Running   0          9m49s
+ds-pipeline-scheduledworkflow-pipelines-definition-7f5c645hcg4g   1/1     Running   0          9m48s
+mariadb-pipelines-definition-795c57795b-27ljh                     1/1     Running   0          9m49s
+----
+
+== Creating a Data Science Pipeline with the KFP SDK
+
+=== Prerequisites 
+* Python Setup
+[source,cmd]
```

Comment:
Move this custom workbench image? Would need to be discussed with the owners of the RHODS administration course.

Reply:
Useful if they're not using notebooks for development